src/components/Sweep.tsx

```diff
@@ -0,0 +1,220 @@
+import { Box, Button, Divider, Flex, Skeleton, Stack, Text as CText } from '@chakra-ui/react'
+import { type AccountId, type AssetId, fromAssetId } from '@shapeshiftoss/caip'
+import type { UtxoBaseAdapter, UtxoChainId } from '@shapeshiftoss/chain-adapters'
+import { bn, FeeDataKey } from '@shapeshiftoss/chain-adapters'
+import type { Utxo } from '@shapeshiftoss/unchained-client/src/generated/bitcoin'
+import { useCallback, useEffect, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { Row } from 'components/Row/Row'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { usePoll } from 'hooks/usePoll/usePoll'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { useGetEstimatedFeesQuery } from 'pages/Lending/hooks/useGetEstimatedFeesQuery'
+import { selectAssetById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { Amount } from './Amount/Amount'
+import { AssetIcon } from './AssetIcon'
+import { handleSend } from './Modals/Send/utils'
+
+const divider = <Divider />
+
+type SweepProps = {
+  assetId: AssetId
+  fromAddress: string | null
+  accountId: AccountId | undefined
+  onBack: () => void
+  onSweepSeen: () => void
+}
+
+const providerIcon = 'https://assets.coincap.io/assets/icons/rune@2x.png'
+
+export const Sweep = ({
+  assetId,
+  fromAddress,
+  accountId,
+  onBack: handleBack,
+  onSweepSeen: handleSwepSeen,
+}: SweepProps) => {
+  const [isSweepPending, setIsSweepPending] = useState(false)
+  const [txId, setTxId] = useState<string | null>(null)
+
+  const {
+    state: { wallet },
+  } = useWallet()
+  const translate = useTranslate()
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+
+  const { data: estimatedFeesData, isLoading: isEstimatedFeesDataLoading } =
+    useGetEstimatedFeesQuery({
+      cryptoAmount: '0',
+      assetId,
+      to: fromAddress ?? '',
+      sendMax: true,
+      accountId: accountId ?? '',
+      contractAddress: undefined,
+      enabled: Boolean(accountId),
+    })
+
+  const handleSweep = useCallback(async () => {
+    if (!wallet) return
+
+    setIsSweepPending(true)
+
+    try {
+      if (!fromAddress || !accountId)
+        throw new Error(`Cannot get from address for accountId: $accountIdcollateralAccountId}`)
```

Comment:
```suggestion
      if (!fromAddress) throw new Error('No fromAddress exists')
      if (!accountId) throw new Error('No accountId exists')
```

Reply:
1511b593e2

Reply:
did you mean to keep the `Cannot get from address for accountId: $accountIdcollateralAccountId}` error?

Reply:
Absolutely - for sweeps, this is required, but it may be undefined on the first render cycles. Should realistically not happen at the point of clicking `handleContinue` but it may. I guess early return is another option here vs. throwing